app/src/main/java/com/framgia/fdms/screen/vendor/VendorViewModel.java

```diff
@@ -116,14 +135,40 @@ public void onLoadVendorFailed() {
 
     @Override
     public void onEditSubmitClick(Producer vendor) {
+        if (TextUtils.isEmpty(vendor.getName()) || TextUtils.isEmpty(vendor.getDescription())) {
+            setMessageError(mActivity.getString(R.string.msg_error_user_name));
+            return;
+        }
         mVendorDialog.dismiss();
+        if (mTypeAction == ACTION_EDIT_VENDOR) {
+            mVendorEdit.setName(vendor.getName());
+            mVendorEdit.setDescription(vendor.getDescription());
+            return;
+        }
+        mVendors.add(0, vendor);
+        mAdapter.notifyItemInserted(0);
+        setPositionScroll(0);
     }
 
     @Override
     public void onEditCancelClick(Producer vendor) {
         mVendorDialog.dismiss();
     }
 
+    @Override
+    public void onAddVendorClick() {
+        mTypeAction = ACTION_ADD_VENDOR;
+        mVendorDialog = VendorDialog.newInstant(this, new Producer());
+        mVendorDialog.show(mActivity.getSupportFragmentManager(),
+            Constant.BundleConstant.BUNDLE_DEVICE);
+    }
+
+    @IntDef({ACTION_EDIT_VENDOR, ACTION_ADD_VENDOR})
+    public @interface Action {
+        int ACTION_EDIT_VENDOR = 0;
+        int ACTION_ADD_VENDOR = 1;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.